.github/workflows/canary-integration-test.yml

```diff
@@ -1760,3 +1760,56 @@ jobs:
         uses: ./.github/workflows/collect-logs
         with:
           name: ${{ github.job }}-${{ matrix.ceph-image }}
+
+  two-object-one-zone:
```

Comment:
Are there any more object store tests we should consider? When there are multiple related scenarios, we do try to be more efficient with the CI resources by running them as separate steps in the same CI job. We just don't want the number of jobs to grow too much over time, and then need to refactor the tests later if runners are too busy.

In this case, you could potentially move the steps in lines 1798-1809 into the other object store job that was just merged? If the pools and object stores are just named differently, hopefully it is simple, though I don't want to cause undue pain in CI troubleshooting. If there is any interaction between the object stores that we want to test independently, then we wouldn't want to combine the steps.

Reply:
Without the CephBlockPool(s), it is a slightly different scenario in that either the CephObjectStore or the CephObjectZone would be creating the pools.  As both of the test scenario have either the pools pre-create by CephBlockPool(s) or CephBlockPool(s) take over after the other controllers have created the pools, I'm not sure if it makes sense to merge them or not.  They are distinct usage modes in my mind.  I doubt pointing two or more CephObjectStore(s) at the same zone is a common deployment model.

Reply:
> I doubt pointing two or more CephObjectStore(s) at the same zone is a common deployment model.

It's not common, but we have had people ask about it. Until now, the use case I've seen has been to have one CephObjectStore for user consumption and one CephObjectStore to do multisite replication. Splitting out for having separate garbage collectors is new to me, but still basically the same idea IMO.

-----

It seems to me that this PR is introducing a CI test that checks 2 things
1. it's possible to predefine pools for object zones
2. it's possible to run multiple object stores per zone where each store has a different purpose

Ideally, it would be nice to see if we could combine this canary test (or the above canary test) with another one either now or at some point soonish. Even though we have more GH action runners than the default, indiscriminate proliferation will cause more CI lag time when multiple PRs are open.

Reply:
IMO, a good follow up might be to combine the test from #14772 with one of the existing canaries like `two-osds-in-device` that currently doesn't even check to see that the child resources it creates become healthy. This would be similar to the `raw-disk-with-object` test but would instead be like `two-osds-in-device-with-custom-object` or something like that.

Reply:
I suppose to that end, we could combine this CI test with `two-osds-in-device` or even `pvc`. Very few of the existing tests really add child complexity. @travisn wdyt? This could spare runners, but it could also cause CI failures for OSD-focused scenarios.

Reply:
I'd like to point out that the easiest way to free up runner resources would be to stop spending several minutes per canary deleting packages from the runner image.

Reply:
> In this case, you could potentially move the steps in lines 1798-1809 into the other object store job that was just merged? If the pools and object stores are just named differently, hopefully it is simple, though I don't want to cause undue pain in CI troubleshooting. If there is any interaction between the object stores that we want to test independently, then we wouldn't want to combine the steps.

I think this would be possible and would be more efficient with CI resources. This would just start up 3 CephObjectStores with 2 effective zones. When there aren't multisite resources, the zone is the same name as the CephObjectStore.

So #14722 would be one CephObjectStore (plus the assumed zone).

Then this test could deploy the 2-stores-1-zone from here, and still test it all in one CI run. It would probably only take an extra minute or two, but it would prevent us from overutilizing runners, which I have noticed can be a problem when we are all rushing to get PRs merged before a big release.

Reply:
> I'd like to point out that the easiest way to free up runner resources would be to stop spending several minutes per canary job deleting packages from the runner image.

We have to do that so that the ceph mons don't become unhealthy from limited disk space.

Reply:
> > I'd like to point out that the easiest way to free up runner resources would be to stop spending several minutes per canary job deleting packages from the runner image.
> 
> We have to do that so that the ceph mons don't become unhealthy from limited disk space.

The disk free threshold can be set via ceph config.

Reply:
> > > I'd like to point out that the easiest way to free up runner resources would be to stop spending several minutes per canary job deleting packages from the runner image.
> > 
> > 
> > We have to do that so that the ceph mons don't become unhealthy from limited disk space.
> 
> The disk free threshold can be set via ceph config.

You're welcome to attempt to set that up in a separate PR. We have spent time investigating how to resolve the mon issue. I believe @subhamkrai did most of that work. For now, this discussion is not germane to what Travis and I are asking for here, which is to try to consolidate the canary tests so we use fewer runners at once.